### ISSUE TITLE
Add max-lines

### ISSUE THREAD

**Issue by pburkindine** [stylelint/stylelint#3808]
> What is the problem you're trying to solve?

Developers create very large S/CSS files which cannot be maintained

> What solution would you like to see?

Rule to set an error when the length of a file exceeds a given integer, same concept as tslint:max-file-line-count

**Comment by jeddy3** [comment]
@pburkindine Thanks for the suggestion.

There is also a [similar rule in ESLint](https://eslint.org/docs/rules/max-lines).

I think we'd want to call the rule `max-lines`. It'll then align with [our naming conventions](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/about-rules.md#about-rule-names) and fit well with the likes of [`max-empty-lines`](https://github.com/stylelint/stylelint/blob/master/lib/rules/max-empty-lines/README.md).

- Name: `max-lines`
- Primary option: `int`
- Secondary options: none
- Message: `Expected no more than ${max} ${max === 1 ? "line" : "lines"}`
- Description: "Limit the number of lines."
- Section: "Limit language features" -> "General / Sheet"

Feel free to [contribute](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#creating-a-new-rule) this rule.

**Comment by Maxim-Mazurok** [comment]
I'm going to work on this one, if nobody else is working already.

**Comment by Maxim-Mazurok** [comment]
I'm thinking about adding `ignore: ["comments", "empty-lines"]` to this new rule. But those options are not requested as of now, but they present in [eslint max-lines rule](https://eslint.org/docs/rules/max-lines). So, @jeddy3, should I include those options, or wait for the request?

**Comment by jeddy3** [comment]
> So, @jeddy3, should I include those options, or wait for the request?

As a rule of thumb, we generally wait until an option is requested before adding it. So, let's just add the plain rule for now.

**Comment by cahamilton** [comment]
Hey there. I hope you don't mind me hijacking this issue - it's a rule I'd like to start enforcing myself in my projects. 

I've pieced together a really basic implementation in #4021, which should cover the basic requirement for enforcing line count (no secondary options). Let me know if theres any issues with it!

**Comment by Maxim-Mazurok** [comment]
@cahamilton, sure, I don't mind at all. I was trying to implement this, but haven't found an optimal way to do this while following general approach used in the code base for other rules.
And then, I switched to other project, without stylelint, so I'm not sure if I'm going to finish this at all.
I will take a look at your PR, thanks!

**Comment by cahamilton** [comment]
@evilebottnawi @hudochenkov @jeddy3 @Maxim-Mazurok

Working through the PR changes now and was wondering, how do we feel about updating the error message to something like the following?

`Expected ${max} ${max === 1 ? "line" : "lines"} max. Actual is ${actual} lines` (or something along those lines). 
 
Would help give more accurate test assertions considering things like different syntaxes, mixed line endings etc. 

Would also be somewhat more inline with error messaging found in ESLint, which uses:
`File must be at most {{max}} lines long. It's {{actual}} lines long.`

**Comment by evilebottnawi** [comment]
Yep, message from eslint looks better

**Comment by jeddy3** [comment]
> Would help give more accurate test assertions considering things like different syntaxes, mixed line endings etc.

That makes sense. How's about we standardise on "Found ..." as it's shorter and a single word e.g. 

```
Expected no more than ${max} ${max === 1 ? "line" : "lines"}. Found ${found} ${found === 1 ? "line" : "lines"} lines.
```

The "Expected ..." convention is shared across all stylelint rules. And the "no more than" is shared across the `*-max-*` rules, so we should stick with them for this rule.

**Comment by pburkindine** [comment]
@jeddy3 @Maxim-Mazurok @cahamilton any update here?

**Comment by evilebottnawi** [comment]
PR welcome, it is open source

**Comment by Maxim-Mazurok** [comment]
No, @pburkindine sorry, I'm not using stylelint currently. Hopefully, @cahamilton will take care of it in #4021

**Comment by jeddy3** [comment]
The community is welcome to [create a plugin for this](https://stylelint.io/developer-guide/plugins).
